File: GATE ONE/DAYSEVEN/PYTHON/DIARY/DiaryMainApplication.py
from Diaries import Diaries

class DiaryMainApplication:
    def __init__(self):
        self.diaries = Diaries()

    @staticmethod
    def display_menu():
        return """\
1. Add Diary.
2. Unlock Diary.
3. Delete Diary.
4. Create Entry
5. Find Entry By ID.
6. Update Entry.
7. Delete Entry.
8. Exit."""

    def run(self):
        while True:
            print(self.display_menu())
            option = input("Select an option: ")

            try:
                option = int(option)
                if option == 1:
                    self.addDiary()
                elif option == 2:
                    self.unlockDiary()
                elif option == 3:
                    self.deleteDiary()
                elif option == 4:
                    self.createEntry()
                elif option == 5:
                    self.findEntry()
                elif option == 6:
                    self.updateEntry()
                elif option == 7:
                    self.deleteEntry()
                elif option == 8:
                    print("Exiting...")
                    break
                else:
                    print("Invalid option")
            except ValueError as e:
                print(f"Error: {e}")

    def addDiary(self):
        username = input("Enter username: ")
        password = input("Enter password: ")
        self.diaries.add(username, password)
        print(f"Diary created for {username}")

    def unlockDiary(self):
        username = input("Enter username: ")
        diary = self.diaries.findByUsername(username)
        if not diary:
            print("Diary not found")
            return
        password = input("Enter password: ")
        diary.unlockDiary(password)

    def deleteDiary(self):
        username = input("Enter username: ")
        password = input("Enter password: ")
        self.diaries.delete(username, password)
        print("Diary deleted")

    def createEntry(self):
        username = input("Enter username: ")
        diary = self.diaries.findByUsername(username)
        if not diary or diary.isLocked():
            print("Diary not found or locked")
            return
        title = input("Enter entry title: ")
        body = input("Enter entry body: ")
        diary.createEntry(title, body)

    def findEntry(self):
        username = input("Enter username: ")
        diary = self.diaries.findByUsername(username)
        if not diary or diary.isLocked():
            print("Diary not found or locked")
            return
        entry_id = int(input("Enter entry ID: "))
        entry = diary.findEntryById(entry_id)
        print(f"Entry found: {entry.getTitle()} {entry.getBody()}")

    def updateEntry(self):
        username = input("Enter username: ")
        diary = self.diaries.findByUsername(username)
        if not diary or diary.isLocked():
            print("Diary not found or locked")
            return
        entry_id = int(input("Enter entry ID: "))
        title = input("Enter new title: ")
        body = input("Enter new body: ")
        diary.updateEntry(entry_id, title, body)

    def deleteEntry(self):
        username = input("Enter username: ")
        diary = self.diaries.findByUsername(username)
        if not diary or diary.isLocked():
            print("Diary not found or locked")
            return
        entry_id = int(input("Enter entry ID: "))
        diary.deleteEntry(entry_id)

if __name__ == "__main__":
    app = DiaryMainApplication()
    app.run()

File: GATE ONE/DAYSEVEN/JAVA/DIARY/DiaryMainApplication.java
import java.util.Scanner;

public class DiaryMainApplication {
    private static final Diaries diaries = new Diaries();
    private static final Scanner input = new Scanner(System.in);

    public static String displayMenu() {
        return """
                1. Add Diary.
                2. Unlock Diary.
                3. Delete Diary.
                4. Create Entry
                5. Find Entry By ID.
                6. Update Entry.
                7. Delete Entry.
                8. Exit.""";
    }

    public static void main(String[] args) {
        while (true) {
            System.out.println(displayMenu());
            System.out.print("Select an option: ");
            int option = input.nextInt();
            input.nextLine();

            try {
                switch (option) {
                    case 1 -> addDiary();
                    case 2 -> unlockDiary();
                    case 3 -> deleteDiary();
                    case 4 -> createEntry();
                    case 5 -> findEntry();
                    case 6 -> updateEntry();
                    case 7 -> deleteEntry();
                    case 8 -> {
                        System.out.println("Exiting...");
                        return;
                    }
                    default -> System.out.println("Invalid option");
                }
            } catch (Exception e) {
                System.out.println("Error: " + e.getMessage());
            }
        }
    }

    private static void addDiary() {
        System.out.print("Enter username: ");
        String username = input.nextLine();
        System.out.print("Enter password: ");
        String password = input.nextLine();
        diaries.add(username, password);
        System.out.println("Diary created for " + username);
    }

    private static void unlockDiary() {
        System.out.print("Enter username: ");
        String username = input.nextLine();
        Diary diary = diaries.findByUsername(username);
        if (diary == null) {
            System.out.println("Diary not found");
            return;
        }
        System.out.print("Enter password: ");
        String password = input.nextLine();
        diary.unlockDiary(password);
    }

    private static void deleteDiary() {
        System.out.print("Enter username: ");
        String username = input.nextLine();
        System.out.print("Enter password: ");
        String password = input.nextLine();
        diaries.delete(username, password);
        System.out.println("Diary deleted");
    }

    private static void createEntry() {
        System.out.print("Enter username: ");
        String username = input.nextLine();
        Diary diary = diaries.findByUsername(username);
        if (diary == null || diary.isLocked()) {
            System.out.println("Diary not found or locked");
            return;
        }
        System.out.print("Enter entry title: ");
        String title = input.nextLine();
        System.out.print("Enter entry body: ");
        String body = input.nextLine();
        diary.createEntry(title, body);
    }

    private static void findEntry() {
        System.out.print("Enter username: ");
        String username = input.nextLine();
        Diary diary = diaries.findByUsername(username);
        if (diary == null || diary.isLocked()) {
            System.out.println("Diary not found or locked");
            return;
        }
        System.out.print("Enter entry ID: ");
        int id = input.nextInt();
        input.nextLine();
        DiaryEntry entry = diary.findEntryById(id);
        System.out.println("Entry found: " + entry.getTitle() + entry.getBody());
    }

    private static void updateEntry() {
        System.out.print("Enter username: ");
        String username = input.nextLine();
        Diary diary = diaries.findByUsername(username);
        if (diary == null || diary.isLocked()) {
            System.out.println("Diary not found or locked");
            return;
        }
        System.out.print("Enter entry ID: ");
        int id = input.nextInt();
        input.nextLine();
        System.out.print("Enter new title: ");
        String title = input.nextLine();
        System.out.print("Enter new body: ");
        String body = input.nextLine();
        diary.updateEntry(id, title, body);
    }

    private static void deleteEntry() {
        System.out.print("Enter username: ");
        String username = input.nextLine();
        Diary diary = diaries.findByUsername(username);
        if (diary == null || diary.isLocked()) {
            System.out.println("Diary not found or locked");
            return;
        }
        System.out.print("Enter entry ID: ");
        int id = input.nextInt();
        input.nextLine();
        diary.deleteEntry(id);
    }
}